lib/SILOptimizer/Mandatory/TFDifferentiation.cpp

```diff
@@ -3516,33 +3587,50 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
     }
   }
 
-  /// Handle `struct_extract` instruction.
-  ///   y = struct_extract <key>, x
-  ///   adj[x] = struct (0, ..., key: adj[y], ..., 0)
   void visitStructExtractInst(StructExtractInst *sei) {
-    auto *structDecl = sei->getStructDecl();
```

Comment:
When we have derived conformances to `Differentiable`, it'd be much more efficient to use the original `struct_extract` differentiation logic when we know that the cotangent type equals the original type and that the property vip is compiler-synthesized (this can be checked via `Decl::isImplicit()`, I think).

Reply:
In fact, this gives me an idea. The original `struct_extract` differentiation logic does not need getter VJPs. So if I can get the original `struct_extract` differentiation logic working on structs with derived conformance to Differentiable (even those with cotangent type not equal to self), then I don't need to write any code that synthesizes getter VJPs.

I think that the original `struct_extract` logic can be made to work even on derived conformances to Differentiable with cotangent type not equal to self, because it should be possible to determine which fields correspond.

I will try out this approach.

Reply:
Sounds reasonable for internal structs. Public structs need property VJPs in any case because of resilience requirements.